firedrake/slate/slac/compiler.py

```diff
@@ -224,7 +229,24 @@ def generate_kernel_ast(builder, statements, declared_temps):
     # Now we wrap up the kernel ast as a PyOP2 kernel and include the
     # Eigen header files
     include_dirs = builder.include_dirs
-    include_dirs.extend(["%s/include/eigen3/" % d for d in PETSC_DIR])
+
+    def extract_eigen_include_dir(PETSC_DIR, PETSC_ARCH):
+        try:
+            with open(os.path.join(PETSC_ARCH, "lib", "petsc", "conf", "petscvariables") if PETSC_ARCH
+                      else os.path.join(PETSC_DIR, "lib", "petsc", "conf", "petscvariables")) as file:
```

Comment:
Prefer:
```python
filepath = os.path.join(PETSC_ARCH or PETSC_DIR, ...)
with open(filepath, "r") as file:
    for line in file:
        if line.find("EIGEN_INCLUDE") == 0:
            ...return ...
    raise RuntimeError("Did not find Eigen include directory in PETSc config %s, do you need --download-eigen?" % filepath)
```

Give that you catch and then immediately re-raise the same exception (filenotfound), there seems no need to catch that one.

Reply:
I was initially trying to make a more explicit error message - left it there after the first refactoring. Is it worth trying to explain why we're trying to open the file ?